core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -0,0 +1,5534 @@
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+
+        case TYPE_BIGNUMBER:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              return convertStringToBinaryString( convertBigNumberToString( (BigDecimal) object ) );
+            case STORAGE_TYPE_BINARY_STRING:
+              String string =
+                  convertBigNumberToString( (BigDecimal) convertBinaryStringToNativeType( (byte[]) object ) );
+              return convertStringToBinaryString( string );
+            case STORAGE_TYPE_INDEXED:
+              return convertStringToBinaryString( convertBigNumberToString( (BigDecimal) index[( (Integer) object )
+                  .intValue()] ) );
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+
+        case TYPE_BOOLEAN:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              return convertStringToBinaryString( convertBooleanToString( (Boolean) object ) );
+            case STORAGE_TYPE_BINARY_STRING:
+              String string = convertBooleanToString( (Boolean) convertBinaryStringToNativeType( (byte[]) object ) );
+              return convertStringToBinaryString( string );
+            case STORAGE_TYPE_INDEXED:
+              return convertStringToBinaryString( convertBooleanToString( (Boolean) index[( (Integer) object )
+                  .intValue()] ) );
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+
+        case TYPE_BINARY:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              return (byte[]) object;
+            case STORAGE_TYPE_BINARY_STRING:
+              return (byte[]) object;
+            case STORAGE_TYPE_INDEXED:
+              return (byte[]) index[( (Integer) object ).intValue()];
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+
+        case TYPE_SERIALIZABLE:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              return convertStringToBinaryString( object.toString() );
+            case STORAGE_TYPE_BINARY_STRING:
+              return (byte[]) object;
+            case STORAGE_TYPE_INDEXED:
+              return convertStringToBinaryString( index[( (Integer) object ).intValue()].toString() );
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+
+        default:
+          throw new KettleValueException( toString() + " : Unknown type " + type + " specified." );
+      }
+    } catch ( ClassCastException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)